azure-core/azure/core/pipeline/policies/universal.py

```diff
@@ -0,0 +1,323 @@
+    def deserialize_from_text(cls, data, content_type=None):
+        # type: (Optional[Union[AnyStr, IO]], Optional[str]) -> Any
+        """Decode data according to content-type.
+
+        Accept a stream of data as well, but will be load at once in memory for now.
+
+        If no content-type, will return the string version (not bytes, not stream)
+
+        :param data: Input, could be bytes or stream (will be decoded with UTF8) or text
+        :type data: str or bytes or IO
+        :param str content_type: The content type.
+        """
+        if hasattr(data, 'read'):
+            # Assume a stream
+            data = cast(IO, data).read()
+
+        if isinstance(data, bytes):
+            data_as_str = data.decode(encoding='utf-8-sig')
+        else:
+            # Explain to mypy the correct type.
+            data_as_str = cast(str, data)
+
+        if content_type is None:
+            return data
+
+        if content_type in cls.JSON_MIMETYPES:
+            try:
+                return json.loads(data_as_str)
+            except ValueError as err:
+                raise DecodeError("JSON is invalid: {}".format(err), error=err)
+        elif "xml" in (content_type or []):
+            try:
+                return ET.fromstring(data_as_str)
+            except ET.ParseError:
+                # It might be because the server has an issue, and returned JSON with
+                # content-type XML....
+                # So let's try a JSON load, and if it's still broken
+                # let's flow the initial exception
+                def _json_attemp(data):
+                    try:
+                        return True, json.loads(data)
+                    except ValueError:
+                        return False, None # Don't care about this one
+                success, json_result = _json_attemp(data)
+                if success:
+                    return json_result
+                # If i'm here, it's not JSON, it's not XML, let's scream
+                # and raise the last context in this block (the XML exception)
+                # The function hack is because Py2.7 messes up with exception
+                # context otherwise.
+                _LOGGER.critical("Wasn't XML not JSON, failing")
+                raise_with_traceback(DecodeError, "XML is invalid")
+        raise DecodeError("Cannot deserialize content-type: {}".format(content_type))
+
+    @classmethod
+    def deserialize_from_http_generics(cls, body_bytes, headers):
+        # type: (Optional[Union[AnyStr, IO]], Mapping) -> Any
+        """Deserialize from HTTP response.
+
+        Use bytes and headers to NOT use any requests/aiohttp or whatever
+        specific implementation.
+        Headers will tested for "content-type"
+        """
+        # Try to use content-type from headers if available
+        content_type = None
+        if 'content-type' in headers:
+            content_type = headers['content-type'].split(";")[0].strip().lower()
```

Comment:
In the case of aiohttp at least, `headers` values are string lists i.e. the `split` has already happened, and so this raises.

Reply:
ooooo good catch. I will update to something like:
Does the `if` statement still work?
If so we could use something like:
```python
if 'content-type' in headers:
    try:
        content_type_header = headers['content-type'].split(";")[0]
    except AttributeError:
        content_type_header = headers['content-type'][0]
    content_type = content_type_header.strip().lower()
```
Would this resolve the errors you're seeing?

Reply:
I think so. It looks feasible to give the response's headers the right shape in `AioHttpTransportResponse.__init__` rather than special-case it here though.